File: javascript/leap/leap.js
//
// This is only a SKELETON file for the "Leap" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

"use strict";

class Year {
  
  constructor (input) {
    this.year = input;
  }
  
  isLeap () {
    const isLeap400 = this.year % 400 === 0;
    const isLeapOther = this.year % 4 === 0 && this.year % 100;
    return (isLeap400 || isLeapOther) ? true : false;
  }
  
}

module.exports = Year;

File: csharp/leap/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        /*
        on every year that is evenly divisible by 4
        except every year that is evenly divisible by 100
        unless the year is also evenly divisible by 400
        */
        var isLeap400 = year % 400 == 0;
        var isLeapOther = year % 4 == 0 && year % 100 != 0;
        return (isLeap400 || isLeapOther) ? true : false;
    }
}